api/v1alpha1/virtualmachine_types.go

```diff
@@ -281,6 +281,20 @@ type VirtualMachineVolumeStatus struct {
 	Error string `json:"error"`
 }
 
+// NetworkInterfaceStatus defines the observed state of the network interfaces desired to be attached
```

Comment:
I would reword this comment to clarify these are the interfaces as seen by vmtools and the guest os. 

(the desired interfaces would be configured in the spec, but it's possible the os would have ignored them, specially in future cases where additional images are supported)

Reply:
sounds good. To verify, Is it okay for us to mention VMware tools in the comments here?.

Reply:
Yeah, I don't see any problem with that

Reply:
I think later we'll have to incorporate and match these up with their corresponding spec entry when possible